guide/concepts/bins.md

```diff
@@ -0,0 +1,35 @@
+# Bins
+
+Bins, similarly to pipelines, are containers for elements. However, at the same time,
+they can be placed and linked within pipelines. Although bin is a separate
+Membrane entity, it can be perceived as a pipeline within an element.
+Bins can be nested within one another.
+
+There are two main reasons why bins are useful:
+- they enable creating reusable element groups,
+- they allow managing elements inside, for instance by dynamically spawning
+or replacing them as the stream changes.
+
+## Pads
+
+Bin's pads are defined similarly to element's pads and can be linked the same way.
+However, their role is limited to proxy the stream to elements inside (inputs)
+or outside (outputs). To achieve that, each input pad of a bin needs to be linked
```

Comment:
Not necessarily, bins can be at the beginning or end of pipeline which means its input/output pads might not be linked with the outside world.

Reply:
Moreover it can be linked to another `child` rather than `element` (we can have multiple bins one after another). See the first comment.

Reply:
Even if a bin is at either end of a pipeline, it is still true that each of its input and output pads is linked.